File: das_backend/src/main/java/ch/sbb/backend/admin/domain/service/DomainServicePointService.java
package ch.sbb.backend.admin.domain.service;

import ch.sbb.backend.admin.domain.ServicePoint;
import ch.sbb.backend.admin.domain.repository.ServicePointRepository;
import java.util.List;
import java.util.Optional;

public class DomainServicePointService implements ServicePointService {

    private final ServicePointRepository servicePointRepository;

    public DomainServicePointService(ServicePointRepository servicePointRepository) {
        this.servicePointRepository = servicePointRepository;
    }

    @Override
    public List<ServicePoint> getAll() {
        return servicePointRepository.findAll();
    }

    @Override
    public Optional<ServicePoint> findByUic(int uic) {
        return servicePointRepository.findByUic(uic);
    }

    @Override
    public void updateAll(List<ServicePoint> servicePoints) {
        servicePointRepository.saveAll(servicePoints);
    }

    @Override
    public void create(ServicePoint servicePoint) {
        servicePointRepository.save(servicePoint);
    }
}

File: backend/src/main/kotlin/ch/sbb/backend/servicepoints/domain/service/DomainServicePointService.kt
package ch.sbb.backend.servicepoints.domain.service

import ch.sbb.backend.servicepoints.domain.ServicePoint
import ch.sbb.backend.servicepoints.domain.repository.ServicePointRepository

class DomainServicePointService(private val servicePointRepository: ServicePointRepository) :
    ServicePointService {
    override fun getAll(): List<ServicePoint> {
        return servicePointRepository.findAll()
    }

    override fun findByUic(uic: Int): ServicePoint? {
        return servicePointRepository.findByUic(uic)
    }

    override fun updateAll(servicePoints: List<ServicePoint>) {
        servicePointRepository.saveAll(servicePoints)
    }

    override fun create(servicePoint: ServicePoint) {
        servicePointRepository.save(servicePoint)
    }

}
